src/code_800AF9B0.c

```diff
@@ -6,6 +6,12 @@
 #include "global.h"
 #include "common_structs.h"
 
+union GrandPrixPointsUnion
+{
+    u8 grandPrixPointsArray[4];
+    u8 grandPrixPointsMushroomCup;
+};
```

Comment:
Here's the union. Let me know if you think there's a better place for it. D_8018ED10 is only referenced in this files so I put it here.

Also if we want to keep the union, I should probably make D_8018ED88 (save data grandPrixPointsBackup) the same type.